_episodes/08-putting-it-all-together.md

```diff
@@ -86,209 +89,226 @@ also change the name of the columns in the DataFrame like this:
+> do as much as you easily can in a convenience layer, i.e. start
+> creating the plot in Pandas or plotnine, and then use matplotlib
+> for the rest.
+{: .callout}
 
-~~~
-    list_numbers = [1.5, 4, 2.2, 5.7]
-    plt.plot(list_numbers)
-    plt.show()
-~~~
-{: .language-python}
+We will cover a few basic commands for creating and formatting plots with matplotlib in this lesson. A great resource for help creating and styling your figures is the matplotlib gallery
+(<http://matplotlib.org/gallery.html>), which includes plots in many different
+styles and the source code that creates them. 
 
-The command `plt.show()` prompts Python to display the figure. Without it, it
-creates an object in memory but doesn't produce a visible plot. The ipython
-notebooks (if using `%matplotlib inline`) will automatically show you the figure
-even if you don't write `plt.show()`, but get in the habit of including this
-command!
+### Using the matplotlib package
 
-If you provide the `plot()` function with only one list of numbers, it assumes
-that it is a sequence of y-values and plots them against their index (the first
-value in the list is plotted at `x=0`, the second at `x=1`, etc). If the
-function `plot()` receives two lists, it assumes the first one is the x-values
-and the second the y-values. The line connecting the points will follow the list
-in order:
+First, import the pyplot toolbox:
 
 ~~~
-    plt.plot([6.8, 4.3, 3.2, 8.1], list_numbers)
-    plt.show()
+import matplotlib.pyplot as plt
 ~~~
 {: .language-python}
 
-A third, optional argument in `plot()` is a string of characters that indicates
-the line type and color for the plot. The default value is a continuous blue
-line. For example, we can make the line red (`'r'`), with circles at every data
-point (`'o'`), and a dot-dash pattern (`'-.'`). Look through the matplotlib
-gallery for more examples.
+By default, matplotlib creates a figure in a separate window. When using
+Jupyter notebooks, we can make figures appear in-line within the notebook by
+writing:
 
 ~~~
-    plt.plot([6.8, 4.3, 3.2, 8.1], list_numbers, 'ro-.')
-    plt.axis([0,10,0,6])
-    plt.show()
+%matplotlib inline
 ~~~
 {: .language-python}
 
-The command `plt.axis()` sets the limits of the axes from a list of `[xmin,
-xmax, ymin, ymax]` values (the square brackets are needed because the argument
-for the function `axis()` is one list of values, not four separate numbers!).
-The functions `xlabel()` and `ylabel()` will label the axes, and `title()` will
-write a title above the figure.
+### `plt` pyplot versus object based matplotlib
 
-A single figure can include multiple lines, and they can be plotted using the
-same `plt.plot()` command by adding more pairs of x values and y values (and
-optionally line styles):
+Matplotlib integrates nicely with the `numpy` package and can use numpy arrays as input of the available plot functions. Consider the following example data, created with `numpy`:
```

Comment:
> Matplotlib works seamlessly

Reply:
Note how we write Matplotlib and numpy. We need agree on how we write all the library names: `library` or Library.

I think it is fine either way, it just has to be consistent. Perhaps, Numpy and Matplotlib ?

Reply:
I addressed pretty much all the comments except this one. I could have sworn matplotlib was never capitalized, but it looks in their documentation like it is, now. It looks like NumPy has also transitioned to the "Py-emphasizing" case. pandas is still lowercase. So I say we go with how the developers do it - Matplotlib, NumPy, pandas.

Thoughts?

Reply:
I agree